Content.Client/CombatMode/ShowCombatModeIndicatorsOverlay.cs

```diff
@@ -0,0 +1,113 @@
+using Content.Shared.CCVar;
+using Content.Client.Hands.Systems;
+using Content.Shared.Weapons.Ranged.Components;
+using Robust.Client.GameObjects;
+using Robust.Client.Graphics;
+using Robust.Client.Input;
+using Robust.Client.UserInterface;
+using Robust.Client.Player;
+using Robust.Shared.Configuration;
+using Robust.Shared.Enums;
+using Robust.Shared.Utility;
+
+namespace Content.Client.CombatMode
+{
+    public sealed class ShowCombatModeIndicatorsOverlay : Overlay
+    {
+        [Dependency] private readonly IConfigurationManager _cfg = default!;
+        [Dependency] private readonly IInputManager _inputManager = default!;
+        [Dependency] private readonly IEntityManager _entMan = default!;
+        [Dependency] private readonly IEyeManager _eye = default!;
+        private Texture? _gunSight;
+        private Texture? _meleeSight;
+        private CombatModeSystem _combatSystem;
+        private IPlayerManager _player;
+        public override OverlaySpace Space => OverlaySpace.ScreenSpace;
+        public ShowCombatModeIndicatorsOverlay(IPlayerManager playerManager, CombatModeSystem combatSys)
+        {
+            IoCManager.InjectDependencies(this);
+
+            _combatSystem = combatSys;
+            _player = playerManager;
+
+            _gunSight = GetTextureFromRsi("gunsight");
+            _meleeSight = GetTextureFromRsi("meleesight");
+        }
```

Comment:
1. Please add spaces
2. You inject some dependencies but pass in IPlayerManager to the ctor. Just pass them all in.
3. Please put all of the dependencies together.
4. Don't make gunsight  / meleesight nullable as they're not intended to be.

Reply:
maybe fixed